code/modules/cooking/machinery/cooking_machines/container.dm

```diff
@@ -0,0 +1,264 @@
+
+/obj/item/reagent_containers/cooking_container
+	icon = 'icons/obj/cooking_machines.dmi'
+	var/shortname
+	var/max_space = 20//Maximum sum of w-classes of foods in this container at once
+	volume = 80//Maximum units of reagents
+	flags = OPENCONTAINER | NOREACT
+	var/list/insertable = list(
+		/obj/item/reagent_containers/food/snacks,
+		/obj/item/holder,
+		/obj/item/paper,
+		/obj/item/flame/candle
+		)
+	var/appliancetype // Bitfield, uses the same as appliances
+	w_class = ITEMSIZE_NORMAL
+
+/obj/item/reagent_containers/cooking_container/examine(var/mob/user)
+	. = ..()
+	if (length(contents))
+		var/string = "It contains:</br><ul><li>"
+		string += jointext(contents, "</li></br><li>") + "</li></ul>"
+		to_chat(user, SPAN_NOTICE(string))
+	if (reagents.total_volume)
+		to_chat(user, SPAN_NOTICE("It contains [reagents.total_volume] units of reagents."))
+
+
+/obj/item/reagent_containers/cooking_container/attackby(var/obj/item/I as obj, var/mob/user as mob)
+	for (var/possible_type in insertable)
+		if (istype(I, possible_type))
+			if (!can_fit(I))
+				to_chat(user, SPAN_WARNING("There's no more space in [src] for that!"))
+				return FALSE
+
+			if(!user.unEquip(I))
+				return
+			I.forceMove(src)
+			to_chat(user, SPAN_NOTICE("You put [I] into [src]."))
+			return
+
+/obj/item/reagent_containers/cooking_container/verb/empty()
+	set src in oview(1)
+	set name = "Empty Container"
+	set category = "Object"
+	set desc = "Removes items from the container, excluding reagents."
+
+	do_empty(usr)
+
+/obj/item/reagent_containers/cooking_container/proc/do_empty(mob/user)
+	if (!isliving(user))
+		//Here we only check for ghosts. Animals are intentionally allowed to remove things from oven trays so they can eat it
+		return
+
+	if (user.stat || user.restrained())
+		to_chat(user, SPAN_NOTICE("You are in no fit state to do this."))
+		return
+
+	if (!Adjacent(user))
+		to_chat(user, "You can't reach [src] from here.")
+		return
+
+	if (isemptylist(contents))
+		to_chat(user, SPAN_WARNING("There's nothing in [src] you can remove!"))
+		return
+
+	for (var/atom/movable/A in contents)
+		A.forceMove(get_turf(src))
+
+	to_chat(user, SPAN_NOTICE("You remove all the solid items from [src]."))
+
+/obj/item/reagent_containers/cooking_container/proc/check_contents()
+	if (length(contents) == 0)
+		if (!reagents || reagents.total_volume == 0)
+			return 0.0//Completely empty
+	else if (length(contents) == 1)
+		if (!reagents || reagents.total_volume == 0)
+			return 1.0//Contains only a single object which can be extracted alone
+	return 2.0//Contains multiple objects and/or reagents
+
+/obj/item/reagent_containers/cooking_container/AltClick(var/mob/user)
+	do_empty(user)
+
+//Deletes contents of container.
+//Used when food is burned, before replacing it with a burned mess
+/obj/item/reagent_containers/cooking_container/proc/clear()
+	for (var/atom/a in contents)
+		qdel(a)
+
+	if (reagents)
```

Comment:
Don't think we need to check if reagents exists, it should exist from Initialization for reagent_containers? Also elsewhere.